### PR TITLE
Create a fresh todo file on every run.

### LINKED ISSUE #916: Tweak the behaviour of how we handle the --todo flag
Coming from #914:

>> I've noticed another strange behavior, though. It seems that when generating the todo file, reek actually respects an existing reek file. Is this intended?

It's obvious that we should tweak the behaviour of how we handle the --todo flag since it's confusing and / or not properly explained.

### I believe we have 2 use cases:

1.) There is already an existing configuration file. 

In this case we could ask the user:

>>
We found the existing configuration file xyz.reek and will take this into account.
Should we append the output of --todo to this file? 
In case of "no" we'll just write the results to STDOUT 
[Yes/no/abort]

In case the user answers "yes", we'll just append the results to the existing configuration file using the existing configuration file with a marker above it like this

```Yaml
    ### Autogenerated by `reek --todo -c xyz.reek` ###
```

In case of "no", we'll just write it to STDOUT and in case of "abort" we'll....just abort.  

2.) There is not existing configuration file

In this case, the behaviour stays as it is right now.

WDYT?

### PR DESCRIPTION
**Pull request by troessner**
Fixes #916 - ready for review!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -404,13 +404,29 @@ via:
 reek -c whatever/.todo.reek lib/
 ```
 
-Note that if you re-run
+It's important to understand that the number one use case of the `--todo` flag
+is to be run once at the beginning of the introduction of Reek to ease the transition.
+If you find yourself re-running Reek with the `--todo` flag in order to silence new warnings
+you're defeating the purpose of both the `--todo` flag and of Reek itself.
+
+As a consequence, this
```
Comment (mvz) [inline README.md]: I would remove the full invocation below, and put something like 'running Reek with the `--todo` flag again'.

File: README.md [hunk 2]
```diff
@@ -404,13 +404,29 @@ via:
...
 
-'.todo.reek' will get overwritten with a possibly updated configuration.
+will overwrite '.todo.reek' without asking (should not be a problem since this file is
+supposed to be versioned) and without taking **any** other configuration file you might
+have into account.
+
+This means that when you run
+
+```Bash
+reek -c config other_configuration.reek --todo lib/
```
Comment (mvz) [inline README.md]: I think the `-c config other_configuration.reek` isn't quite correct.

**Comment by mvz** [comment]
Review done!

File: README.md (hunk 2 above)
Comment (troessner) [inline README.md]: Good catch, fixed!

File: README.md (hunk 1 above)
Comment (troessner) [inline README.md]: Ack, fixed.

**Comment by troessner** [comment]
Addressed all comments - lgtm? :)

**Comment by mvz** [comment]
More tweaks of the README may result from our discussion in #916, but looks good for now. Merging.

**Pull request merged**